engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/util/CollectionUtil.kt

```diff
@@ -19,15 +19,15 @@ inline fun <reified T> Collection<*>.singleOfType(predicate: (T) -> Boolean = {
     return singleOfTypeOrNull(predicate) as T
 }
 
-inline fun <K, E> Iterable<E>.toMap(crossinline keyExtractor: (E) -> K): Map<K, E> {
+inline fun <K, E> Iterable<E>.strictAssociateBy(crossinline keyExtractor: (E) -> K): Map<K, E> {
```

Comment:
I realize that Kotlin already has a `associateBy` extension function that does something similar. So I renamed this to better match that.